content/blog/2024/11/how-to-create-an-esp-idf-component/index.md

```diff
@@ -0,0 +1,483 @@
+{{< /alert >}}
+
+
+Now it's time to start with the new component. For this article, we will use an I2C temperature and humidity sensor.
+
+### The sensor: SHTC3
+
+For this new component, the sensor will be the [Sensirion SHTC3](https://sensirion.com/products/catalog/SHTC3). This is a I2C temperature and humidity sensor with a typical accuracy of ±2 %RH and ±0.2 °C. To know more about the specifications, please visit the official [product page](https://sensirion.com/products/catalog/SHTC3).
+
+If you don't have this sensor, you can change to any other I2C device. All the information you might need for the new sensor should be provided by the manufacturer, including the device address, registers, etc.
+
+### The Board: ESP32-C3-DevKit-RUST-1
+
+We chose the sensor SHTC3, because it can be found on the [ESP32-C3-DevKit-RUST-1](https://github.com/esp-rs/esp-rust-board/tree/v1.2) development board. That is why, this is the board that we will use for developing the component.
+
+
+The **ESP32-C3-DevKit-RUST-1** development board has 2 sensors connected to the I2C peripheral on the following GPIOs:
+
+| Signal     | GPIO        |
+|------------|-------------|
+| SDA        | GPIO10      |
+| SCL        | GPIO8       |
+
+All the project files for this board are available on GitHub.
+
+
+{{< github repo="esp-rs/esp-rust-board" >}}
+
+### Creating the component
+
+Before creating the component, we will need to create a new project that the component will be part of. If you have already the project, you can skip this step and jump to the component creation.
+
+The component can be created manually or you can use the `idf.py` tool to create all the basic skeleton. We recommend to create the new component by the tool provided by Espressif.
+
+To create the component, we will go through some steps:
+
+1. Create a new project.
+2. Create the new component inside the project.
+3. Write the driver code for the selected sensor.
+
+> We assume that you already have the ESP-IDF version 5.2 or higher installed on your system.
+
+#### Additional content from DevCon23
+
+On the Espressif DevCon23, Ivan Grokhotkov, gave a talk titled: Developing, Publishing, and Maintaining Components for ESP-IDF. You can watch this talk as an additional material for your studies.
+
+{{< youtube D86gQ4knUnc >}}
+
+The Component Manager [documentation](https://docs.espressif.com/projects/idf-component-manager/en/latest/) is very comprehensive and you will find all the information you need there, including the [simple component](https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html#a-simple-esp-idf-component) structure.
+
+#### Create the new project
+
+To create the component, we will use a project as the starting point. This project will be used to test the component.
+
+```bash
+idf.py create-project example-shtc3
+```
+
+Now inside the project folder, let's test the build for the ESP32-C3.
+
+```bash

```

Comment:
If you are planning to show how to upload the component to the registry in the next one, then it's better to make the structure inside out. I.e. `shtc3` component > `examples` > `example-shtc3` 

And use the `override_path` in the example's manifest.

Reply:
I'll do it in the next one. I want to keep this as simple as possible.